src/main/java/pl/javastart/task/Truck.java

```diff
@@ -0,0 +1,34 @@
+package pl.javastart.task;
+
+public class Truck extends Car {
+    private int weight;
+
+    public Truck(String name, double tankCapacity, double averagePetrolConsumption, boolean isAcOn, int weight) {
+        super(name, tankCapacity, averagePetrolConsumption, isAcOn);
+        this.weight = weight;
+    }
+
+    public int getWeight() {
+        return weight;
+    }
+
+    public void setWeight(int weight) {
+        this.weight = weight;
+    }
+
+    @Override
+    public double range() {
+        return (getTankCapacity() * 100) / computeAveragePetrolConsumption();
+    }
```

Comment:
Ogólny algorytm się nie zmienia i zasięg to zawsze "pojemność baku / aktualne spalanie * 100km". Jedyne co się zmienia, to aktualne spalanie, czyli metoda `computeAveragePetrolConsumption`